lib/jsdom/events/EventException.js

```diff
@@ -0,0 +1,27 @@
+﻿"use strict";
+
+var inheritFrom = require("../utils").inheritFrom;
+
+function EventException() {
+  if (arguments.length > 0) {
+    this._code = arguments[0];
+  } else {
+    this._code = 0;
+  }
+  if (arguments.length > 1) {
+    this._message = arguments[1];
+  } else {
+    this._message = "Unspecified event type";
+  }
+  Error.call(this, this._message);
+  if (Error.captureStackTrace) {
+    Error.captureStackTrace(this, EventException);
+  }
+}
+
+inheritFrom(Error, EventException, {
+  UNSPECIFIED_EVENT_TYPE_ERR: 0,
+  get code() { return this._code; }
+});
+
+module.exports.EventException = EventException;
```

Comment:
We should probably have a unified approach across all objects for this, in case "private" (jsdom-exclusive) methods will be added later. I'd have the object exported as exports[objectName], then all private methods can go on the exports too.

Another approach would be having a $private-Symbol, which we can require everywhere and just set on the constructor with an object which includes our private methods.